code/game/objects/structures/tables_racks.dm

```diff
@@ -65,17 +43,25 @@
 		COMSIG_ATOM_ENTERED = PROC_REF(on_cross),
 		COMSIG_ATOM_EXIT = PROC_REF(on_try_exit),
 		COMSIG_OBJ_TRY_ALLOW_THROUGH = PROC_REF(can_climb_over),
+		COMSIG_FIND_FOOTSTEP_SOUND = TYPE_PROC_REF(/atom/movable, footstep_override),
+		COMSIG_TURF_CHECK_COVERED = TYPE_PROC_REF(/atom/movable, turf_cover_check),
 	)
 	AddElement(/datum/element/connect_loc, connections)
 
-/obj/structure/table/proc/on_cross(datum/source, atom/movable/O, oldloc, oldlocs)
-	SIGNAL_HANDLER
-	if(!istype(O,/mob/living/carbon/xenomorph/ravager))
-		return
-	var/mob/living/carbon/xenomorph/M = O
-	if(!M.stat) //No dead xenos jumpin on the bed~
-		visible_message(span_danger("[O] plows straight through [src]!"))
-		deconstruct(FALSE)
+/obj/structure/table/Destroy()
+	update_adjacent(loc) //so neighbouring tables get updated correctly
+	return ..()
+
+/obj/structure/table/deconstruct(disassembled)
+	if(disassembled)
+		new parts(loc)
+	else
+		if(reinforced)
+			if(prob(50))
+				new /obj/item/stack/rods(loc)
+		if(dropmetal)
+			new sheet_type(src)
+	return ..()
```

Comment:
early returns?

Reply:
? it needs to be after though

Reply:
? it needs to be after though

Reply:
I mean you do x
Return ..()
Do x 
Return ..()
Instead of if Elaine

Reply:
For the disassembled check

Reply:
Oh sorry